Added missing model documentation for various Cloud models (#5914)
nautobot/nautobot: nautobot/docs/user-guide/core-data-model/cloud/cloudservice.md

# Cloud Services

> In Nautobot, cloud services enable tracking of resources, applications, and data storage provided to user via remote servers. A cloud service represents a specific instance, within a user's [cloud network](./cloudnetwork.md), of a service defined by its [cloud type](./cloudtype.md). For instance, it could describe a server instance like "c7gn.xlarge" from Amazon EC2 or a Linux virtual machine instance named "M416ms_v2" created using Microsoft Azure Virtual Machine Service within the user's virtual private cloud.

Reviewer: I would think that `c7gn.xlarge` itself would be a cloud type, rather than a cloud service - an EC2 instance using that cloud type (a specific VM) would be a cloud service.

> Each cloud service must be assigned to a [cloud type](./cloudtype.md) and a [cloud network](./cloudnetwork.md) and can be optionally assigned to a [cloud account](./cloudaccount.md). Extra configurations can be stored in JSON format in the `extra_config` attribute.

Reviewer: Can we elaborate on when/why you would/wouldn't associate a CloudService directly to a CloudAccount? (IIRC it's because the CloudType already has an assigned CloudAccount in most cases? But please sanity-check me on this.)

Author: `CloudType` does not have a `CloudAccount` associated with it, only provider(manufacturer), however, `CloudNetwork` does have a mandatory `CloudAccount` attribute. I will take note of that.

# Cloud Services

In Nautobot, cloud services enable tracking of resources, applications, and data storage provided to user via remote servers. A cloud service represents a specific instance, within a user's [cloud network](./cloudnetwork.md), of a service defined by its [cloud type](./cloudtype.md). For example, it could describe an EC2 instance of type "c7gn.xlarge" from Amazon EC2 or a Linux virtual machine instance of type "M416ms_v2" created using Microsoft Azure Virtual Machine Service within the user's virtual private cloud.

Each cloud service must be assigned to a [cloud type](./cloudtype.md) and a [cloud network](./cloudnetwork.md) and can be optionally assigned to a [cloud account](./cloudaccount.md). Extra configurations can be stored in JSON format in the `extra_config` attribute.

!!! note

>     Note that a [cloud network](./cloudnetwork.md) already has a cloud acoount [cloud account](./cloudaccount.md) associated with it. So in most cases, the `cloud_account` attribute of cloud service instances can be left blank. You can assign a cloud account to a cloud service when you have different cloud accounts for your cloud network and cloud service.

Reviewer: ```suggestion
    Note that a [cloud network](./cloudnetwork.md) already has a [cloud account](./cloudaccount.md) associated with it. So in most cases, the `cloud_account` attribute of cloud service instances can be left blank. You can assign a cloud account to a cloud service to handle any case where the service belongs to a different account than its associated network.
```

The word `cloud` is beginning to lose all meaning after that last sentence. :-)
